### PR TITLE
Specify content for all requests

### LINKED ISSUE #598: Content of requests in section 7.0 not clear
It's not explicitly or clearly stated what the **content** of requests to the Statement API should look like. 

In particular, when POSTing a single Statement, does that Statement need to be in an array? I'd say yes, but I don't see that stated anywhere.

### PR DESCRIPTION
**Pull request by garemoko**
Fixes #598

### CODE REVIEW

**Comment by andyjohnson** [comment]
Picky, but I don't like "Content" - I'd think these should either be "Request Content" and "Return Content"  (also okay with Values instead of Content or whatever people think makes sense) or change "Content" to "Contains" so both it and "Returns" are verbs.

**Comment by garemoko** [comment]
I like "Request Content" and "Return Content", but can we agree it before I go through and change them all? I'll likely use find and replace and then check it, but even so!

**Comment by garemoko** [comment]
Now also addresses point 3 in #544

File: xAPI.md [hunk 1]
```diff
@@ -2996,7 +2998,9 @@ on the AP, and have the LRS query that API for the list of updated (or new)
 Statements periodically. This will likely only be a realistic option for systems 
 that provide a lot of data to the LRS.  
 
-Returns: ```200 OK```, Array of Statement id(s) (UUID).  
+**Content:** An array of Statements or a single Statement object to be stored.
```
Comment (fugu13) [inline xAPI.md]: It should either be "array of Statement objects" or "a single Statement to be stored", not the mixture of nomenclature it is now. I'm fine either way.

File: xAPI.md [hunk 2]
```diff
@@ -3406,8 +3412,11 @@ Example endpoint: http://example.com/xAPI/activities/state
 Stores, changes, fetches, or deletes the document specified by the given stateId that 
 exists in the context of the specified Activity, Agent, and registration (if specified).  
 
-Returns (PUT | POST | DELETE): ```204 No Content```  
-Returns (GET): ```200 OK```, State Content  
+**Content (PUT | POST):** The Document object to be stored or updated.
```
Comment (fugu13) [inline xAPI.md]: Isn't necessarily an object (with PUT); we generally just call them documents, so propose we continue to do so (lowercase).

File: xAPI.md [hunk 3]
```diff
@@ -3544,8 +3559,11 @@ Example endpoint: http://example.com/xAPI/activities/profile
 Stores, changes, fetches, or deletes the specified profile document in the context of the 
 specified Activity.  
 
-Returns (PUT | POST | DELETE): ```204 No Content```  
-Returns (GET): ```200 OK```, Profile Content  
+**Content (PUT | POST):** The Document object to be stored or updated.
```
Comment (fugu13) [inline xAPI.md]: same object issue as above

File: xAPI.md [hunk 4]
```diff
@@ -3406,8 +3412,11 @@ Example endpoint: http://example.com/xAPI/activities/state
 Stores, changes, fetches, or deletes the document specified by the given stateId that 
 exists in the context of the specified Activity, Agent, and registration (if specified).  
 
-Returns (PUT | POST | DELETE): ```204 No Content```  
-Returns (GET): ```200 OK```, State Content  
+**Content (PUT | POST):** The Document object to be stored or updated.  
+**Content (GET | DELETE):** None.  
+
+**Returns (PUT | POST | DELETE):** ```204 No Content```  
+**Returns (GET):** ```200 OK```, State Content
```
Comment (fugu13) [inline xAPI.md]: I think we should just call say ", the State document"

File: xAPI.md [hunk 5]
```diff
@@ -3544,8 +3559,11 @@ Example endpoint: http://example.com/xAPI/activities/profile
 Stores, changes, fetches, or deletes the specified profile document in the context of the 
 specified Activity.  
 
-Returns (PUT | POST | DELETE): ```204 No Content```  
-Returns (GET): ```200 OK```, Profile Content  
+**Content (PUT | POST):** The Document object to be stored or updated.  
+**Content (GET | DELETE):** None.  
+
+**Returns (PUT | POST | DELETE)** ```204 No Content```  
+**Returns (GET):** ```200 OK```, Profile Content
```
Comment (fugu13) [inline xAPI.md]: "the Profile document"?

File: xAPI.md [hunk 6]
```diff
@@ -3703,8 +3725,11 @@ Example endpoint: http://example.com/xAPI/agents/profile
 Stores, changes, fetches, or deletes the specified profile document in the context of the 
 specified Agent.  
 
-Returns (PUT | POST | DELETE): ```204 No Content```  
-Returns (GET): ```200 OK```, Profile Content  
+**Content (PUT | POST):** The Document object to be stored or updated.  
+**Content (GET | DELETE):** None.  
+
+**Returns (PUT | POST | DELETE):** ```204 No Content```  
+**Returns (GET):** ```200 OK```, Profile Content
```
Comment (fugu13) [inline xAPI.md]: "the Profile document" as above

File: xAPI.md [hunk 7]
```diff
@@ -3703,8 +3725,11 @@ Example endpoint: http://example.com/xAPI/agents/profile
 Stores, changes, fetches, or deletes the specified profile document in the context of the 
 specified Agent.  
 
-Returns (PUT | POST | DELETE): ```204 No Content```  
-Returns (GET): ```200 OK```, Profile Content  
+**Content (PUT | POST):** The Document object to be stored or updated.
```
Comment (fugu13) [inline xAPI.md]: object issue as above

File: xAPI.md [hunk 8]
```diff
@@ -3768,7 +3795,9 @@ allow other uses to emerge.
 
 Example endpoint: http://example.com/xAPI/about
 
-Returns: ```200 OK```, Single 'about' JSON document.
+**Content:** None.
+
+**Returns:** ```200 OK```, Single 'about' JSON document.
```
Comment (fugu13) [inline xAPI.md]: "single" feels weird, as does using document here. "JSON object with basic metadata about this LRS"?

File: xAPI.md (hunk 1 above)
Comment (garemoko) [inline xAPI.md]: deleting "object"

File: xAPI.md (hunk 8 above)
Comment (garemoko) [inline xAPI.md]: How about 'containing' instead of 'with'?

```JSON object containing basic metadata about this LRS```

**Comment by garemoko** [comment]
@fugu13 please check the latest commit addresses your comments.

File: xAPI.md (hunk 8 above)
Comment (fugu13) [inline xAPI.md]: works for me

**Comment by fugu13** [comment]
:+1:

**Comment by garemoko** [comment]
Andrew to fix merge conflict.

**Comment by andyjohnson** [comment]
Have 2 +1s to merge after @garemoko fixes

**Comment by garemoko** [comment]
This should be good to merge now

**Pull request merged**